### PR TITLE
Improve compatibility with cmake 3.5 and QT 5.5-5.11.

### LINKED ISSUE #1772: Mention CUTTER_USE_BUNDLED_RADARE2 in build documentation.
For development builds using bundled radare2 option seems like the easiest and most convenient way to me. It saves the trouble with choosing method for building r2, it overriding system's r2  or configuring build system to recognize install location if it was configured not to override system's r2.

### LINKED ISSUE #1796: cmake 3.5.2 configuration fails
(continued from [here](https://github.com/radareorg/cutter/pull/1553#issuecomment-533916719), text copied below)

**Environment information**
* Operating System: Slackware 14.2
* Cutter version: pulled latest from repo (a15d104b4559310cfe9603f6f979496656c1adf9)

**Describe the bug**
`cmake` 3.5.2 fails even if the docs say that >3.1 should be ok

**To Reproduce**
Try to build with a `cmake` < 3.7 :grin:

I have `cmake` version 3.5.2 and it fails since it does not know about `VERSION_GREATER_EQUAL` - apparently that was introduced in version 3.7.
Google told me that I could change the line
```
if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.12.0")
```
in `CMakeMakeLists.txt` with
```
if (POLICY CMP0074)
```
With this change `cmake` goes a little further, then fails again but I think for some unrelated reason.

### PR DESCRIPTION
**Pull request by karliss**
**Detailed description**

Goal is being able to compile on Ubuntu 16.04 (currently supported LTS version) using the version of tools and library available in official repository.

* Minor Qt api changes
* Stack widget converted to custom model, prevents the need for event loop hack.
* Some of the common sync/unsync code for memory widgets moved to base class, sync/unsync action added to context menu that opens when clicking on memory dock header.
* Refactoring of graph context menu and other mouse event handling. Simplify common code. Use proper Qt context menu event instead checking for right click mouse press, this prevents some of the problems with duplicate context menu and in future might help with non mouse based context menu opening.

**Known limitations**

I don't think it's worth fixing most of them.
* Update worker not supported - depends on https://doc.qt.io/qt-5/qversionnumber.html
* OpenGL graph drawing backend not supported on  < Qt 5.6 due to dependency on OpenGLExtraFunctions. Regular graph drawing mode works.
* Error reporting with Breakpad causes linker error. Most likely due to CMake differences. Should be cleaned up regardless of CMake 3.5 support but that's a work for separate issue.
* Worse highdpi support and dock management in older Qt versions. It is somewhat expected that newer versions of library improve things and if you want to use those new features you need a newer version of library.
* find_pkg based inclusion of r2 doesn't work CMake < 3.6 due to missing IMPORTED_TARGET workaround is to use one of the other methods for detecting r2 like bundled r2 option.

**Test plan (required)**

* Added new Travis task
  * 0eb74db demonstrates that it can detect problems caused by features missing in older qt versions
  * cmake version can be seen in log for 49215eb which has debug code, in later versions indirectly reported by meson from r2 build
* manually tested in a Ubuntu 16.04 VM with Qt and CMake  from default repositories, meson needs to be  installed using pip

**Closing issues**

Closes #1796, closes #1772

### CODE REVIEW

**Comment by karliss** [comment]
- [ ] Update worker
- [ ] StackWidget
- [ ] graph sync/unsync
- [ ] investigate color scheme problems

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -106,16 +105,7 @@ endif()
...
-        pkg_check_modules(GVC IMPORTED_TARGET libgvc)
-        if (GVC_FOUND)
-            set(CUTTER_ENABLE_GRAPHVIZ ON)
-        else()
-            set(CUTTER_ENABLE_GRAPHVIZ OFF)
-        endif()
-    else()
-        pkg_check_modules(GVC REQUIRED IMPORTED_TARGET libgvc)
-    endif()
+    include(FindGraphviz)
```
Comment (thestr4ng3r) [inline src/CMakeLists.txt]: FindGraphviz.cmake should be a find module exporting a target like `Graphviz::GVC` and used with `find_package(Graphviz ...`, see https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html#a-sample-find-module
We could also use an existing one, but unfortunately many modules out there don't follow modern cmake standards.

Comment (karliss) [reply src/CMakeLists.txt]: Yeah will do that, forgot about FindX cmake files having special meaning. Aliases should allow pkg_check IMPORTED_TARGET to be exposed the same way as manually created one. What do you mean by existing one?

Comment (thestr4ng3r) [reply src/CMakeLists.txt]: I meant copying one that has already been written, like this: https://github.com/KDAB/KDStateMachineEditor/blob/master/cmake/FindGraphviz.cmake
But this one is ugly and doesn't export a target.

**Pull request merged**